spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/core/PubSubConfiguration.java

```diff
@@ -92,11 +147,27 @@ public Subscriber getSubscriber(String name, String projectId) {
    * @param subscriptionName subscription name
    * @param projectId project id
    * @return flow control settings
+   * @deprecated use {@link #computeSubscriberFlowControlSettings(ProjectSubscriptionName)}
    */
+  @Deprecated
   public FlowControl computeSubscriberFlowControlSettings(
       String subscriptionName, String projectId) {
-    FlowControl flowControl = getSubscriber(subscriptionName, projectId).getFlowControl();
+    return computeSubscriberFlowControlSettings(ProjectSubscriptionName.of(projectId, subscriptionName));
+  }
+
+  /**
+   * Computes flow control settings to use. The subscription-specific property takes precedence if
+   * both global and subscription-specific properties are set. If subscription-specific settings are
+   * not set then global settings are picked.
+   *
+   * @param psn Fully qualified subscription name
+   * @return flow control settings defaulting to global where not provided
+   */
+  public FlowControl computeSubscriberFlowControlSettings(ProjectSubscriptionName psn) {
```

Comment:
a minor suggestion here: 
```suggestion
  public FlowControl computeSubscriberFlowControlSettings(ProjectSubscriptionName projectSubscriptionName) {
```